File: day1/solution.hs
import Control.Monad
import System.IO

countIncrements::[Integer]->Integer
countIncrements nums= sum [ if y>x then 1 else 0 | (x,y)<- zip nums (tail nums)]

makeSliding:: [Integer] -> [Integer]
makeSliding nums =[x+y+z | (x,y,z)<- zip3 nums (tail nums) (tail $ tail nums)]

main::IO ()
main = do 
     contents <- readFile "input.txt" 
     let nums= read <$> lines contents
     print $ countIncrements  nums
     print $ countIncrements $ makeSliding nums

File: day1/solution.py
with open("input.txt", "r") as infile:
    data= infile.read()

nums= list(map(int,data.split()))
def countIncrements(nums):
    count=0
    for (x,y) in zip(nums,nums[1:]):
        if (y>x):
            count+=1
    return count

def makeSliding(nums):
    return [x+y+z for x,y,z in zip (nums, nums [1:],nums[2:])]


print("#1st solution=",countIncrements(nums))
print("#2nd solution=",countIncrements(makeSliding(nums)))


